src/components/markdown-renderer/replace-components/csv/csv-replacer.tsx

```diff
@@ -0,0 +1,27 @@
+import { DomElement } from 'domhandler'
+import React from 'react'
+import { ComponentReplacer } from '../ComponentReplacer'
+import { CsvTable } from './csv-table'
+
+export class CsvReplacer implements ComponentReplacer {
+  getReplacement (codeNode: DomElement, index: number): React.ReactElement | undefined {
+    if (codeNode.name !== 'code' || !codeNode.attribs || !codeNode.attribs['data-highlight-language'] || codeNode.attribs['data-highlight-language'] !== 'csv' || !codeNode.children || !codeNode.children[0]) {
+      return
+    }
+
+    const code = codeNode.children[0].data as string
+
+    const extraData = codeNode.attribs['data-extra']
+    const extraInfos = /\s*(delimiter=([^\s]*))?\s*(header)?/.exec(extraData)
```

Comment:
Please extract this regex into a const outside of the function.

Reply:
done